harmonizer/src/lib.rs

```diff
@@ -55,13 +61,59 @@ pub fn harmonize_limit(
     // The snapshot is created in the build_harmonizer.rs script and included in our binary image
     let buffer = include_bytes!(concat!(env!("OUT_DIR"), "/composition.snap"));
 
+    let initial_heap_size =
+        std::env::var("APOLLO_ROUTER_HARMONIZER_EXPERIMENTAL_V8_INITIAL_HEAP_SIZE").unwrap_or_else(|_e| {
+            APOLLO_ROUTER_HARMONIZER_EXPERIMENTAL_V8_INITIAL_HEAP_SIZE_DEFAULT.to_string()
+        });
+
+    let max_heap_size =
+        std::env::var("APOLLO_ROUTER_HARMONIZER_EXPERIMENTAL_V8_MAX_HEAP_SIZE").unwrap_or_else(|_e| {
+            APOLLO_ROUTER_HARMONIZER_EXPERIMENTAL_V8_MAX_HEAP_SIZE_DEFAULT.to_string()
+        });
```

Comment:
This doesn't run in Router-land, so we should remove `_ROUTER` from these and just go with `APOLLO_HARMONIZER_EXPERIMENTAL_V8_MAX_HEAP_SIZE`

Reply:
sounds good, I actually ran some additional tests and it looks like `--max-heap-size` becomes useless as soon as resizing is involved, i.e it's possible to go beyond max-heap-size through resizing. I'm thinking to update the logic so heap resizing only happens if max-heap-size env variable is not provided.

Reply:
Updated the PR to only resize if max heap size has been explicitly set